finance/management/commands/xformnetfilerawdata.py

```diff
@@ -0,0 +1,365 @@
+            if past_money.count() > 0:
+                # Figure it out from past contributions.
+                ballot_item_response = past_money[0] \
+                    .beneficiary.ballot_item_response
+            else:
+                # Figure out beneficiary from item text.
+                try:
+                    ballot_item_response = parse_candidate_info(
+                        row, ballot=ballot)
+                except:
+                    ballot_item_response = parse_referendum_info(
+                        row, ballot=ballot)
+
+            return ballot_item_response, True
+        beneficiary.ballot_item_response, beneficiary.support = \
+            parse_ballot_info(row)
+        beneficiary.save()
+
+        # Now we have all the parts, save!
+        money = models.IndependentMoney(  # or calculated_Amount
+            amount=row['tran_Amt1'],
+            report_date=date_parse(row['tran_Date']),
+            reporting_period=reporting_period,
+            benefactor_zip=bf_zip_code,
+            benefactor=benefactor,
+            beneficiary=beneficiary,
+            source='NF',
+            source_xact_id=row['netFileKey'])
+        money.save()
+
+        if self.verbosity:
+            print(str(money))
+
+    def get_committee_benefactor(self, row):
+        """ Utility function to identify a committee benefactor.
+        """
+        return models.CommitteeBenefactor.objects.get_or_create(
+            name=row['tran_NamL'].strip(),
+            filer_id=self.clean_filer_id(row.get('cmte_Id', '')))[0]
+
+    def get_committee_beneficiary(self, row):
+        """ Utility function to identify a committee beneficiary.
+        """
+        filer_id = row['filerId']
+        if '-' in filer_id:
+            filer_id = '-'.join(filer_id.split('-')[1:])
+        else:
+            filer_id = filer_id
+
+        return models.Beneficiary.objects.get_or_create(
+            name=row['filerName'].strip(), filer_id=filer_id)[0]
+
+    @staticmethod
+    def clean_filer_id(filer_id):
+        """ Utility function to scrub committee IDs."""
+        if isinstance(filer_id, Number):
+            return str(int(filer_id))
+        elif np.any([filer_id.startswith(c) for c in ('C', '#')]):
+            filer_id = filer_id[1:]
+        return filer_id or None  # don't do blank.
+
+    def load_f497_data(self):
+        """ Loads data from Form 460 Schedule A:
+        contributions to primarily formed committees."""
+        # Split data by type, add column defs.
+        self.f497_data = self.data[self.data['form_Type'] == 'F497P2']
+        if self.verbosity:
+            print("NYI: load %d rows of form 497 data." % len(self.f497_data))
+
+    def load_f496_data(self):
+        # Split data by type, add column defs.
+        self.f496_data = self.data[self.data['form_Type'] == 'F496P3']
+        if self.verbosity:
+            print("NYI: load %d rows of form 496 data." % len(self.f496_data))
```

Comment:
So, uh, that's a lot of code. Since testing is cool, would you mind adding one or two tests of code in this module? That is, just to get the framework in place, and then we can fill in more tests later. At least, it always seems to me like writing the first test is the hardest, so I want to saddle you with that burden if possible :grinning: